sim-cli/src/main.rs

```diff
@@ -70,43 +71,74 @@ async fn main() -> anyhow::Result<()> {
             )));
         }
 
-        if r_alias_map.contains_key(&node_info.alias) {
+        if alias_node_map.contains_key(&node_info.alias) {
             anyhow::bail!(LightningError::ValidationError(format!(
                 "duplicated node: {}.",
                 node_info.alias
             )));
         }
 
-        r_alias_map.insert(node_info.alias.clone(), node_info.pubkey);
         clients.insert(node_info.pubkey, node);
+        pk_node_map.insert(node_info.pubkey, node_info.clone());
+        alias_node_map.insert(node_info.alias.clone(), node_info);
     }
 
     let mut validated_activities = vec![];
     // Make all the activities identifiable by PK internally
     for act in activity.iter_mut() {
         // We can only map aliases to nodes we control, so if either the source or destination alias
         // is not in r_alias_map, we fail
-        if let NodeId::Alias(a) = &act.source {
-            if let Some(pk) = r_alias_map.get(a) {
-                act.source = NodeId::PublicKey(*pk);
-            } else {
-                anyhow::bail!(LightningError::ValidationError(format!(
-                    "unknown activity source: {}.",
-                    act.source
-                )));
+        let source = if let Some(source) = match &act.source {
+            NodeId::PublicKey(pk) => pk_node_map.remove(pk),
+            NodeId::Alias(a) => alias_node_map.remove(a),
+        } {
+            source
+        } else {
+            anyhow::bail!(LightningError::ValidationError(format!(
+                "unknown activity source: {}.",
+                act.source
+            )));
+        };
+
+        let destination = match &act.destination {
+            NodeId::Alias(a) => {
+                if let Some(info) = alias_node_map.remove(a) {
+                    info
+                } else {
+                    anyhow::bail!(LightningError::ValidationError(format!(
+                        "unknown activity destination: {}.",
+                        act.destination
+                    )));
+                }
             }
-        }
-        if let NodeId::Alias(a) = &act.destination {
-            if let Some(pk) = r_alias_map.get(a) {
-                act.destination = NodeId::PublicKey(*pk);
-            } else {
-                anyhow::bail!(LightningError::ValidationError(format!(
-                    "unknown activity destination: {}.",
-                    act.destination
-                )));
+            NodeId::PublicKey(pk) => {
+                if let Some(info) = pk_node_map.remove(pk) {
```

Comment:
Do we need to remove here? As is, config with multiple activity descriptions for a single source node will fail.

I think that we want to be able to support activity of this nature: 
```
Alice -> Dave / 10 sat every 5 mins
Alice -> Bob / 1000 sat every 30 mins
```

Reply:
Oh damn, that was me trying to be clever by avoiding copying the `NodeInfo` :D

This actually applies to both repeated sources and destinations. My bad.